controller_interface/include/controller_interface/controller_interface.hpp

```diff
@@ -96,9 +96,15 @@ class ControllerInterface : public rclcpp_lifecycle::node_interfaces::LifecycleN
   CONTROLLER_INTERFACE_PUBLIC
   virtual return_type update(const rclcpp::Time & time, const rclcpp::Duration & period) = 0;
 
+  CONTROLLER_INTERFACE_PUBLIC
+  void set_node(std::shared_ptr<rclcpp_lifecycle::LifecycleNode>);
+
   CONTROLLER_INTERFACE_PUBLIC
   std::shared_ptr<rclcpp_lifecycle::LifecycleNode> get_node();
 
+  CONTROLLER_INTERFACE_PUBLIC
+  std::shared_ptr<rclcpp_lifecycle::LifecycleNode> get_node_const() const;
```

Comment:
Why this?

Reply:
`joint_trajectory_controller.cpp` wasn't building because some of it's const methods couldn't guarantee the node wouldn't be modified without a const getter function. Another option would be to make those function not const. I'm also open to other ideas. The error was:
```
/ros2_controllers/joint_trajectory_controller/src/joint_trajectory_controller.cpp:1130:16: error: passing ‘const joint_trajectory_controller::JointTrajectoryController’ as ‘this’ argument discards qualifiers [-fpermissive]
 1130 |       get_node()->get_logger(),
```

Reply:
Is it possible to have both version of the `get_node()` method? `const` and non-const. I think it would be possible to have the same names still. This would lead to clearer API

Reply:
That is a lot cleaner, and it looks like it worked. I'll update it.